src/book/02-system/01-nubots/03-nuclear.mdx

```diff
@@ -5,3 +5,144 @@ title: NUClear
+| Always   | Used to request continuous reactions in the system. `on<Always>` will ensure a single instance of the associated reaction is running at all times.                                                                                                                                                                                                                                                                                             |
+| Startup  | Used to specify reactions which should occur at startup. `on<Startup>()` will create a task after all reactors have been installed into the PowerPlant but before the system starts the main execution phase.                                                                                                                                                                                                                                      |
+| Shutdown | Used to specify any reactions/tasks which should occur during shutdown. `on<Shutdown>()` will create a task after the shutdown command has been emitted and all existing tasks have completed.                                                                                                                                                                                                                                                |
+| IO       | Used to trigger reactions based on standard I/O operations using file descriptors. `on<IO>(pipe/stream/comms, io_status)` will create a task when the communication line matches the given status. For example, `on<IO>(my_file_descriptor, IO::READ)` will create the task when the given file descriptor has data available to read.                                                                                                                                                                                                                                                                |
+| Network  | NUClear provides a networking protocol to send messages to other devices on the network. `on<Network<T>>()` will create a task when T is emitted using the emission type `Scope::NETWORK`.                                                                                                                                                                                                                                                                 |
+
+More information on DSL words for on statements can be found in the [NUClear documentation](https://nuclear.readthedocs.io/en/latest/dsl.html#on-statements).
+
+### Runtime arguments
+
+Some DSL words will provide the ability to make changes to the system during runtime. This means that NUClear avoids the need for a system restart should a configuration, port number, or file need to be changed while the system is running.
+
+The DSL words that take runtime arguments are: `IO`, `TCP`, and `UDP`.
```

Comment:
Since this is a NUbots specific page, we should probably also include `Configuration` and also include it in the dsl words above